homeassistant/components/vartastorage/sensor.py

```diff
@@ -0,0 +1,221 @@
+                    "unit_of_measurement": "W",
+                }
+                thisdict[8] = {
+                    "name": "VARTA charge power",
+                    "state": result.charge_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[9] = {
+                    "name": "VARTA discharge power",
+                    "state": result.discharge_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[10] = {
+                    "name": "VARTA error code",
+                    "state": result.error_code,
+                    "device_class": "None",
+                    "state_class": "None",
+                    "unit_of_measurement": "",
+                }
+                thisdict[11] = {
+                    "name": "VARTA production power",
+                    "state": result.production_power,
+                    "device_class": "power",
+                    "state_class": "measurement",
+                    "unit_of_measurement": "W",
+                }
+                thisdict[12] = {
+                    "name": "VARTA total production power",
+                    "state": result.total_production_power,
+                    "device_class": "energy",
+                    "state_class": "total_increasing",
+                    "unit_of_measurement": "kWh",
+                }
+                thisdict[13] = {
+                    "name": "VARTA total charged energy",
+                    "state": result.total_charged_energy,
+                    "device_class": "energy",
+                    "state_class": "total",
+                    "unit_of_measurement": "kWh",
+                }
+
+                return thisdict
+        except Exception as failed_update:
+            raise UpdateFailed("Error communicating with API") from failed_update
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=1),
+    )
+
+    #
+    # Fetch initial data so we have data when entities subscribe
+    #
+    # If the refresh fails, async_config_entry_first_refresh will
+    # raise ConfigEntryNotReady and setup will try again later
+    #
+    # If you do not want to retry setup on failure, use
+    # coordinator.async_refresh() instead
+    #
+    await coordinator.async_config_entry_first_refresh()
+
+    async_add_entities(
+        MyEntity(coordinator, idx) for idx, ent in enumerate(coordinator.data)
+    )
+
+
+class MyEntity(CoordinatorEntity, SensorEntity):
```

Comment:
I suggest renaming the class to something more custom to your entity (`VartaStorageEntity`?)

Reply:
done